presto-elasticsearch/src/main/java/com/facebook/presto/elasticsearch/ElasticsearchTableDescriptionProvider.java

```diff
@@ -29,60 +29,39 @@
 import java.util.List;
 import java.util.Map;
 import java.util.Set;
-import java.util.function.Supplier;
 
 import static com.facebook.presto.elasticsearch.ElasticsearchErrorCode.ELASTICSEARCH_TABLE_DEFINITION_ERROR;
 import static com.google.common.base.MoreObjects.firstNonNull;
 import static com.google.common.base.Preconditions.checkArgument;
 import static com.google.common.base.Strings.isNullOrEmpty;
 import static java.nio.file.Files.readAllBytes;
-import static java.util.Objects.requireNonNull;
 
-public class ElasticsearchTableDescriptionSupplier
-        implements Supplier<Map<SchemaTableName, ElasticsearchTableDescription>>
+public class ElasticsearchTableDescriptionProvider
 {
-    private final ElasticsearchConnectorConfig config;
-    private final JsonCodec<ElasticsearchTableDescription> codec;
+    private final Map<SchemaTableName, ElasticsearchTableDescription> tableDefinitions;
 
     @Inject
-    ElasticsearchTableDescriptionSupplier(ElasticsearchConnectorConfig config, JsonCodec<ElasticsearchTableDescription> codec)
+    ElasticsearchTableDescriptionProvider(ElasticsearchConnectorConfig config, JsonCodec<ElasticsearchTableDescription> codec)
     {
-        this.config = requireNonNull(config, "config is null");
-        this.codec = requireNonNull(codec, "codec is null");
-    }
-
-    @Override
-    public Map<SchemaTableName, ElasticsearchTableDescription> get()
-    {
-        return createElasticsearchTableDescriptions(config.getTableDescriptionDir(), config.getDefaultSchema(), ImmutableSet.copyOf(config.getTableNames()), codec);
-    }
-
-    public static Map<SchemaTableName, ElasticsearchTableDescription> createElasticsearchTableDescriptions(
```

Comment:
I think this method was OK, what do you think?

```
    ElasticsearchTableDescriptionProvider(ElasticsearchConnectorConfig config, JsonCodec<ElasticsearchTableDescription> codec)
{ 
    requireNonNull(config, "config is null");
    requireNonNull(codec, "codec is null");
    this. tableDefinitions =  createElasticsearchTableDescriptions(...);
}
```
Then we could have public methods at the top.